packages/teleport/src/Nodes/AddNode/useAddNode.ts

```diff
@@ -14,65 +14,62 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { useState, useEffect } from 'react';
-import { formatDistanceStrict } from 'date-fns';
+import { useState } from 'react';
 import useAttempt from 'shared/hooks/useAttemptNext';
 import TeleportContext from 'teleport/teleportContext';
-import { BashCommand } from 'teleport/services/nodes';
 import cfg from 'teleport/config';
-import { JoinToken } from 'teleport/services/joinToken';
+import { JoinToken, Rule } from 'teleport/services/joinToken';
 
 export default function useAddNode(ctx: TeleportContext) {
-  const { attempt, run } = useAttempt('processing');
+  const { attempt, run } = useAttempt('');
   const isEnterprise = ctx.isEnterprise;
   const version = ctx.storeUser.state.cluster.authVersion;
   const user = ctx.storeUser.state.username;
   const isAuthTypeLocal = !ctx.storeUser.isSso();
-  const [automatic, setAutomatic] = useState(true);
-  const [script, setScript] = useState('');
-  const [expiry, setExpiry] = useState('');
-  const [token, setToken] = useState('');
-
-  useEffect(() => {
-    createJoinToken();
-  }, []);
+  const [method, setMethod] = useState<JoinMethod>('iam');
+  const [token, setToken] = useState<JoinToken>(null);
+  const [iamJoinToken, setIamJoinToken] = useState<JoinToken>(null);
```

Comment:
nit: i think setting it to nothing is equivalent to null (or i think more accurately undefined), unless for some reason we want it to specifically be null?
```suggestion
  const [token, setToken] = useState<JoinToken>();
  const [iamJoinToken, setIamJoinToken] = useState<JoinToken>();
```